### ISSUE TITLE
Fatal Error on submit booking request in admin bookings

### ISSUE THREAD

**Issue by ajgw** [woothemes/woocommerce-accommodation-bookings#25]
Latest version of extension 1.0.2 from this morning:

I had a booking request with order. In bookings list in admin I saw the eye symbol and and a checkmark symbol for this booking inline. After clicking on checkmark symbol making this booking a true booking, I got a fatal error. Instead of sending a email to customer, I saw the output on screen with this error message in email body:

Product	Quantity	Price
FeWo Alte Liebe auf Juist mieten
Check-in at 16:00 Uhr 
Check-out 
Fatal error: Uncaught Error: Call to a member function add() on boolean in /wp-content/plugins/woocommerce-accommodation-bookings/includes/class-wc-accommodation-booking-order-info.php:64 Stack trace: #0 /wp-content/plugins/woocommerce-accommodation-bookings/includes/class-wc-accommodation-booking-order-info.php(44): WC_Accommodation_Booking_Order_Info->get_end_date_timestamp(NULL, 0) #1 /wp-includes/plugin.php(525): WC_Accommodation_Booking_Order_Info->add_checkinout_info_to_order_email(3, Array, Object(WC_Order)) #2 /wp-content/plugins/woocommerce/templates/emails/email-order-items.php(45): do_action('woocommerce_ord...', 3, Array, Object(WC_Order)) #3 /wp-content/plugins/woocommerce/includes/wc-core-functions.php(199): include('/www...') #4 /wp-content/plugins/woocommerce/includes/abstracts/abstr in /wp-content/plugins/woocommerce-accommodation-bookings/includes/class-wc-accommodation-booking-order-info.php on line 64

**Comment by gedex** [comment]
@ajgw 1.0.2 is not released yet. Are you checking out from `master` branch?

**Comment by ajgw** [comment]
Yes.

Creating a manual booking worked fine for me – you are on your way. Thanks.

**Comment by gedex** [comment]
Interesting. What's the start date and end date for this accommodation-booking, @ajgw ? Also the date format you use in your site.

From your fatal error, following code:

~~~php
DateTime::createFromFormat( get_option( 'date_format' ) . ' H:i:s', $start_date . ' 00:00:00' );
~~~

returns `false`.

**Comment by ajgw** [comment]
Well date format is j. F Y. Start date is 10. September 2016, 16:00 Uhr and end date is 17. September 2016, 10:00 Uhr or in english date format 09-10-2016 til 09-17-2016.

**Comment by gedex** [comment]
Using inputs provided from you:

~~~php
DateTime::createFromFormat( 'j. F Y H:i:s', '10. September 2016 00:00:00' );
~~~

it returns `DateTime` object as expected:

```
object(DateTime)#1146 (3) {
  ["date"]=>
  string(19) "2016-09-10 00:00:00"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(3) "UTC"
}
```

Could you give me the screenshot of order items in edit order? Something like:

![screenshot](https://cldup.com/Z88n39Qvrx.png)

**Comment by ajgw** [comment]
![bildschirmfoto 2016-02-12 um 13 51 37](https://cloud.githubusercontent.com/assets/3390880/13007318/c90d3e5c-d18f-11e5-8e6a-b0bc4d26eeb6.png)

**Comment by ajgw** [comment]
Beside: you see at line duration the word nights after number? This string is not translatable in pot or .po file – didn't found string. Is it coming from Javascript?

**Comment by gedex** [comment]
@ajgw Nope, not from JS. It's duration unit from product type. But I agree with you it should be translatable. Will log that into another issue.

**Comment by ajgw** [comment]
This is a screenshot from output I got after clicked on Icon:

![bildschirmfoto 2016-02-12 um 11 51 28](https://cloud.githubusercontent.com/assets/3390880/13007507/1ab603fa-d191-11e5-9af9-801c8dba2368.png)

**Comment by solaceten** [comment]
I was also able to recreate this bug with 1.2

`Check-in at 3:00 pm
Check-out
Fatal error: Call to a member function add() on a non-object in /home/[removed]/public_html/_TEST/wp-content/plugins/woocommerce-accommodation-bookings-master/includes/class-wc-accommodation-booking-order-info.php on line 64`

**Comment by gedex** [comment]
Ok. I looked again at the original message:

> I had a booking request with order. In bookings list in admin I saw the eye symbol and and a checkmark symbol for this booking inline. After clicking on checkmark symbol making this booking a true booking, I got a fatal error

So, this is accommodation booking product which requires confirmation. I tried to reproduce this:

* As logged-in customer, I requested the booking.
* I checked the booking in booking list, it has "Pending Confirmation" status, as shown below:

  ![screenshot-1](https://cldup.com/v2UqxP17R0.png)

* I confirmed the booking by clicking the checkmark button as @ajgw pointed
* I got the booking confirmation email without error (also no fatal error in the logs)

   ![screenshot-2](https://cldup.com/PgW4rauX4r.png)

So I couldn't reproduce the issue. However, I'll do check on `DateTime::createFromFormat` when it returns `false`. Looking at your email:

```
Check-in at 16:00 Uhr
```

The check-in date is missing. So, order item meta `Booking Date` probably doesn't exists or empty. And guess what? It's localised one when saved by Bookings. Will fix this.

**Comment by gedex** [comment]
@ajgw @solaceten Please pull and test.

**Comment by solaceten** [comment]
Confirm this appears to fix it - thanks

**Comment by ajgw** [comment]
Fine, you are the best. It works, I got the email with all details.

But wait: I set my previously confirmed booking back to "not confirmed" and saved booking. Got a fatal error:

Warning: in_array() expects parameter 2 to be array, null given in /wp-content/plugins/woocommerce-bookings/includes/admin/class-wc-bookings-details-meta-box.php on line 449

Warning: Cannot modify header information - headers already sent by (output started at /wp-content/plugins/woocommerce-bookings/includes/admin/class-wc-bookings-details-meta-box.php:449) in /www/htdocs/w00ffce5/fewo-christiane/wp-includes/pluggable.php on line 1228

But switch from confirmed to not confirmed was saved. Error only on saving booking.

**Comment by solaceten** [comment]
I was not able to re-produce this error!  Made a new booking, changed status, updated - all status changes worked fine.

**Comment by ajgw** [comment]
It is maybe related to translation?

**Comment by gedex** [comment]
This error:

```
Warning: in_array() expects parameter 2 to be array, null given in /wp-content/plugins/woocommerce-bookings/includes/admin/class-wc-bookings-details-meta-box.php on line 449

Warning: Cannot modify header information - headers already sent by (output started at /wp-content/plugins/woocommerce-bookings/includes/admin/class-wc-bookings-details-meta-box.php:449) in /www/htdocs/w00ffce5/fewo-christiane/wp-includes/pluggable.php on line 1228
```

comes from Bookings. @ajgw I just logged the issue in Bookings repo, will be fixed in Bookings 1.9.6. And yes, it's the same issues -- translation string when checking 'Booking ID' in order item meta.
